pkg/utils/resources.go

```diff
@@ -0,0 +1,74 @@
+package utils
+
+// Inspired from SNR - https://github.com/medik8s/self-node-remediation/blob/main/controllers/selfnoderemediation_controller.go#L283-L346
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	log = ctrl.Log.WithName("utils-resource")
+)
+
+func DeleteResources(r client.Client, nodeName string) error {
```

Comment:
we need to pass the context from the reconciler as parameter here, and use it in all client calls